### PR TITLE
ED-2366 ExRed home page tests

### PR DESCRIPTION
**Pull request by kowalcj0**
This [ticket](https://uktrade.atlassian.net/browse/ED-2366)

Apart from implementing the scenarios this PR includes basic project setup and Docker configuration.

Scenario:
```gherkin
  @ED-2366
  @sections
  Scenario: Any Exporter should see the "Video, Exporting Journey, Personas, Guidance, Services, Case Studies" sections on the home page.
      Given "Robert" visits the "Home" page
      Then "Robert" should see the "Video, Exporting Journey, Personas, Guidance, Services, Case Studies" sections on Home page


  @ED-2366
  @triage
  Scenario: Any Exporter visiting the home page should be able to get to triage
    Given "Robert" visits the "Home" page for the first time

    When "Robert" decides to get started in Exporting journey section

    Then "Robert" should be on the "Triage - 1st question" page


  @ED-2366
  @triage
  Scenario: Any Exporter visiting the home page after triage should be able to get to personalised page
    Given "Robert" has answered triage questions
    And "Robert" goes to the "home" page again

    When "Robert" decides to continue in Exporting journey section

    Then "Robert" should be on the "Personalised Journey" page


  @wip
  @ED-2366
  @personas
  @articles
  Scenario Outline: Any Exporter should get to a relevant article list from Personas section on the homepage
    Given "Robert" classifies himself as "<exporter_status>" exporter

    When "Robert" goes to the relevant "<exporter_status>" exporter link in the Personas section on the homepage

    Then "Robert" should see an ordered list of "previous + next 5" articles selected for "<exporter_status>" exporter
    And "Robert" should see a Articles Read counter, Total number of Articles and Time to complete remaining chapters

    Examples:
      | exporter_status |
      | New             |
      | Occasional      |
      | Regular         |
```

### CODE REVIEW

File: docker-compose-exred.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+services:
+
+  exred_tests:
+    build:
+      context: .
+      dockerfile: Dockerfile-exred
+    working_dir: /usr/src/app
+    command: ./docker/cmd-exred.sh
+    env_file: ./docker/.env
+    volumes:
```
Comment (konradko) [inline docker-compose-exred.yml]: nice

File: docker/create_envs.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash -xe
 
-python ./docker/env_writer.py ./docker/env.json
\ No newline at end of file
+python ./docker/env_writer.py $1
```
Comment (konradko) [inline docker/create_envs.sh]: I think the reason for this script was to have it all in one place - now we reference `create_envs.sh` in makefile, which references cmd line argument in it. Not a big deal, just not sure if better, while a bit less readable. 

Maybe we should just put it in makefile as `DOCKER_COMPOSE_CREATE_ENVS := python ./docker/env_writer.py ./docker/env.json`? Needs checking if the script is not used elsewhere

File: makefile [hunk 3]
```diff
@@ -134,4 +134,40 @@ docker_integration_tests: docker_remove_all
...
+
+EXRED_SET_DOCKER_ENV_VARS := \
+	export EXRED_TESTS_EXRED_UI_URL=https://dev.exportreadiness.directory.uktrade.io
+	export EXRED_TESTS_CIRCLE_SHA1=$(CIRCLE_SHA1)
+
+EXRED_SET_LOCAL_ENV_VARS := \
+	export EXRED_UI_URL=https://dev.exportreadiness.directory.uktrade.io
+
+EXRED_DOCKER_COMPOSE_CREATE_ENVS := \
+	./docker/create_envs.sh ./docker/env_exred.json
```
Comment (konradko) [inline makefile]: yeah no need for shell - the python script can be used

File: docker/create_envs.sh (hunk 2 above)
Comment (kowalcj0) [reply docker/create_envs.sh]: I wanted to keep this sub-project as independent as possible from `smoke` & `functional` tests. Thus it was rather a reasonable decision to have a separate `env.json` file.
I like the latter option.

File: requirements_exred.ini [hunk 4]
```diff
@@ -0,0 +1,5 @@
+behave==1.2.5
+paver==1.2.4
+requests==2.11.1
+selenium==3.6.0
+-e git+https://git@github.com/uktrade/directory-constants.git@v5.5.0#egg=directory-constants
```
Comment (konradko) [inline requirements_exred.ini]: We need to get rid of `-e` in our requirements everywhere, that's 'dev' mode for pip - not downsides AFAIK, but we should be able to just `git+https://git@github.com/uktrade/directory-constants.git@v5.5.0#egg=directory-constants`

File: tests/exred/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+* python 3.5
+* pip
+* docker
+* docker-compose
+
+
+## Run tests locally
+
+Run all scenarios on [BrowserStack](https://www.browserstack.com/automate) in parallel:
```
Comment (konradko) [inline tests/exred/README.md]: maybe mention credentials and where to get them

File: tests/exred/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+## Run tests in Docker container
+
+To run all tests in a container:
+
+```bash
+make exred_docker_tests
+```
+
+
+# TODO
```
Comment (konradko) [inline tests/exred/README.md]: that's for JIRA, not readme

File: requirements_exred.ini (hunk 4 above)
Comment (kowalcj0) [reply requirements_exred.ini]: `pip-compile` It doesn't work without `-e` ! :cry:

File: tests/exred/pages/personalised_journey.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+GUIDANCE_SECTION = {
+    "guidance - heading": "#resource-guidance h2",
+    "guidance - introduction": "#resource-guidance p.section-intro",
+    "guidance - categories": "#resource-guidance div.group",
+}
+
+EXPECTED_ELEMENTS = {}
+# ARTICLES_SECTION is not included as it's not displayed to Regular Exporters
+EXPECTED_ELEMENTS.update(HERO_SECTION)
+# EXPECTED_ELEMENTS.update(FACTS_SECTION)
```
Comment (konradko) [inline tests/exred/pages/personalised_journey.py]: uh oh

File: tests/exred/README.md (hunk 6 above)
Comment (kowalcj0) [reply tests/exred/README.md]: done

File: tests/exred/README.md (hunk 5 above)
Comment (kowalcj0) [reply tests/exred/README.md]: See line 16 - `[BrowserStack](https://www.browserstack.com/users/sign_up) account (DIT employees can find them in Rattic)`

File: requirements_exred.ini (hunk 4 above)
Comment (konradko) [reply requirements_exred.ini]: ah, that's the reason why we have it :(

**Comment by kowalcj0** [comment]
Will merge it despite few failing scenarios (caused by issues with ExRed back-end)

**Pull request merged**